### PR TITLE
Appends a hint if length is expected but hint provided.

### LINKED ISSUE #564: Hint when number is provided instead of length / angle / ...
The call `#set text(size: 12)` should yield a hint along with its error. It could be something like `A length needs a unit – did you mean 12pt?`

Submitted via user feedback.

### PR DESCRIPTION
**Pull request by astrale-sharp**
Closes #564

Adds a test that should pass but doesn't cause of the test infrastructure.

Note that the test currently fails due to the test infrastructure most likely, how to handle newline in errors?

The hint currently is appended to the error, ask if you want a different behavior.

### CODE REVIEW

File: src/eval/cast.rs [hunk 1]
```diff
@@ -291,6 +292,14 @@ impl CastInfo {
             msg.push_str(", found ");
             msg.push_str(found.type_name());
         }
+        if_chain::if_chain! {
+            if let Value::Int(i) = found;
+            if parts.contains(&EcoString::from(Value::Length(Length::zero()).type_name()));
```
Comment (astrale-sharp) [inline src/eval/cast.rs:297]: A bit ugly, some advice?

File: tests/typ/meta/diag.typ [hunk 2]
```diff
@@ -0,0 +1,7 @@
+// Test diagnostics.
+// Ref: false
+
+---
+// Error: 1:17-1:19 expected length, found integer
```
Comment (astrale-sharp) [inline tests/typ/meta/diag.typ:5]: This newline is the reason the test is failing, the test infrastructure could be changed or the message replaced by the hint?

Comment (laurmaedje) [reply tests/typ/meta/diag.typ:5]: for now i would just write something like "expected length, found integer: a length needs a unit – did you mean 12pt?" long-term we can have a more elaborate diagnostics structure with hints

File: src/eval/cast.rs (hunk 1 above)
Comment (laurmaedje) [inline src/eval/cast.rs:297]:
```suggestion
            if parts.iter().any(|p| p == Length::TYPE_NAME);
```
maybe `p.as_str()` if necessary

Comment (astrale-sharp) [reply src/eval/cast.rs]: I was looking for `Length::TYPE_NAME` thanks!

**Comment by astrale-sharp** [comment]
Probably ready to go then!

File: src/eval/cast.rs [hunk 3]
```diff
@@ -291,6 +293,14 @@ impl CastInfo {
             msg.push_str(", found ");
             msg.push_str(found.type_name());
         }
+        if_chain::if_chain! {
+            if let Value::Int(i) = found;
+            if parts.iter().any(|p| p == Length::TYPE_NAME);//.contains(&EcoString::from(Value::Length(Length::zero()).type_name()));
```
Comment (laurmaedje) [inline src/eval/cast.rs:298]:
```suggestion
            if parts.iter().any(|p| p == Length::TYPE_NAME);
```

**Pull request merged**